### DISCUSSION TITLE
The best approach to use the current image URL

### DISCUSSION THREAD

**Discussion by Sergey98Am** [supabase/supabase#21236] [Questions] [answered]
For my React app, I use local and remote Supabase projects. Of course, local db has its storage, and remote has its own.
• Local image URL example: `http://127.0.0.1:54321/storage/v 1/object/public/genre-images/ 45cf53372ec7d06e4e2671c64ad4b14c.png`
• Remote image URL example: `https://ejompcachdnibrtnrnts.supabase.co/storage/v1/object/public/genre-images/45cf53372ec7d06e4e2671c64ad4b14.png`
For example, if I send data from the remote database to the local with `npx supabase do dump
--data-only -f supabase/seed.sql` command, the image column URL is started with `https:// ejompcachdnibrtnrnts.supabase.co`, but in the
local repository, I want to use the local image URL automatically, without manually changing the remote image URLs to the local URLs in a local database. I think it's good to have local variables for local and remote storage URIs, but I'm not sure it's best practice or not in this case. What can I do in this case?

**Comment by GaryAustin1** [comment, marked as answer]
Why not store just the path (or bucket and path) then use getPublicUrl to get the url?  It does not make a database call (just does string concatenation) and uses your supabase-js setup to get the actual instance url.

**Reply by Sergey98Am** [reply]
@GaryAustin1  Everything is running smoothly. Thank you very much for your assistance. I have two questions; one of them is unrelated to Supabase. If it's not too much trouble, I'd greatly appreciate your advice.

1. Is it best practice to store only the image name in Supabase column, rather than the complete URL? Is that a common practice?
2. In my code below, using 'const' inside 'map' works very well and is highly convenient for me. However, is this a common practice? Is it OK to use?

```
<div>
    {newMovies?.map((movie) => {
        // Movie image with optimization
        const { data: movieImage } = supabase.storage
            .from("movie-images")
            .getPublicUrl(movie.image, {
                transform: {
                    width: 400,
                },
            });

        return (
            // Movie item
            <div
                className="w-[200px] lg:w-1/5"
                key={movie.id}
            >
                <MediaCard
                    to={`movie/${movie.id}`}
                    variant="releasedTime"
                    image={movieImage.publicUrl}
                    releasedTime={movie.created_at}
                />
            </div>
        );
    })}
</div>;
```

**Reply by GaryAustin1** [reply]
I can't comment on best practices in the industry.  But not storing the full url is fairly common as you don't want to have to redo the entire database if the URL for the site changes.

I don't have an opinion on using const in a map function.  Should be info on the web as that is JS.

**Reply by Sergey98Am** [reply]
@GaryAustin1 Thank you so much
